experiments/add_sf_to_df_add_row.R

```diff
@@ -168,6 +173,56 @@ geo_attributes <- function(dat, zoomto = NULL, col_add = TRUE){
                          data = data_copy,
                          zoom_to = zoomto)
 
+    #need to somehow update the list 'original_sf' on the fly
+    # making the original_sf list reactive to the df$data seems to work
+    # still testing...
+
+    original_sf_rec <- reactive({
+
+      og_sf <- df$data %>%
+        dplyr::mutate(geo_type = as.character(st_geometry_type(.)))
+
+      og_sf <- st_sf(og_sf, crs = APP_CRS)
+      og_sf <- split(og_sf , f = og_sf$geo_type)
+      og_sf
+    }
+    )
+
+    sf_reac <- reactive({
+
+      og_sf <- df$data %>%
+        dplyr::mutate(geo_type = as.character(st_geometry_type(.)))
+
+      og_sf <- st_sf(og_sf, crs = APP_CRS)
+      og_sf <- split(og_sf , f = og_sf$geo_type)
+      og_sf
+
+
+    })
+
+
+    observe({
+      edits <- callModule(
+        editMod,
+        leafmap = {if (!is.null(original_sf)) {
+          mapv <- mapview(original_sf_rec(), color = 'red', col.regions = 'red', alpha.regions = 0.2)@map
+        } else if ('sf' %in% class(dat)){
+          print('sf')
+          mapv <- mapview(color = 'red', col.regions = 'red', alpha.regions = 0.2)@map %>% leafem::addFeatures(sf_reac(), layerId = sf_reac()$leaf_id, group = 'editLayer')
```

Comment:
Hi @tim-salabim, 

Sorry to reach out to you in this way but I am a bit stuck. We (@joshualerickson)  and I are getting pretty close with this geo_attributes function. But we are having one final issue; when the input is an sf object the edit option does not work. I think it is because of the way we are instantiating the map, I suspect I am getting the group and layer ids wrong.  

I have been trying to emulate the method used in editFeatures.sf but without success https://github.com/r-spatial/mapedit/blob/master/R/edit.R#L283

So here, the leaf_id is the unique row number, and I am assigning the group number to "editLayer" then below (row 219)  I am assigning the targetLayerId the "editLayer". 

Can you advise if I am heading in the right direction, or if I am just completely wrong.

thanks

Matt

Reply:
Hey @mrjoh3, just added some updates for using sf as input. In short, I'm not completely sure if it's sustainable but it somehow works on my end... Since we are using row_order as leaf_id when we add an sf, this can then be cross-referenced with 'layerId' attribute via mapedit. The other part is adding leaflet as the mapping when an sf is added instead of mapview. The few things I'd like to work on tomorrow are; add 'edit_toolbar' functionality (right now i've only explored deleting), when inputting multiple geoms (poly, point, etc) the code I pushed seems to duplicate those geometries 'behind the scenes'. Thanks again for all your help with this! I thought about editing/deleting sf's and thought, 'no way, can't do it' but then I saw what you were doing and was 'ahhhh, nice'! I think it can be done now, hahaha! Have a good one.

Reply:
Hey @mrjoh3 @tim-salabim, seems to be working really good EXCEPT with 'sfc_geometries'. Been a real doozy trying to get leafem::addFeatures() and 'sfc_geoms' to work succinctly. However, if you stick with one geometry it works really smooth! I'll keep working on this week but going to be in the field for a few days so most likely not going to touch until the weekend, thanks.

Reply:
Hey @joshualerickson just tried it on a different computer and it also worked. So I am very excited. I will have some time to work on it this weekend. I will focus on cleaning the process up and moving the function into the package for the pull request into mapedit.